sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/JoinOptimizationSuite.scala

```diff
@@ -116,7 +127,12 @@ class JoinOptimizationSuite extends PlanTest {
     )
 
     queryAnswers foreach { queryAnswerPair =>
-      val optimized = Optimize.execute(queryAnswerPair._1.analyze)
+      val optimized = Optimize.execute(queryAnswerPair._1.analyze) match {
+        // `ReorderJoin` may add `Project` to keep the same order of output attributes.
+        // So, we drop a top `Project` for tests.
+        case project: Project => project.child
```

Comment:
I'm a little hesitate to change this, because if we really forget to add a Project node after join reordering, the test can pass, but that's wrong.

Reply:
yea, great suggestion and I think so. I'll try to fix.